### PR TITLE
Jessxdavid auth persistence

### LINKED ISSUE #11: Add Database Persistence to Next Auth using Prisma Adapter
### Description

Currently, we use next auth without any persistent storage of user identity. This isn't great, because we want user accounts to be persisted in our database so that when our backend needs to restart or crashes, we don't lose any data!

`next-auth` has some great docs on how we can get this working with Prisma. Just follow the docs (linked below).

https://next-auth.js.org/adapters/prisma

### Verification Steps

Log into the site as a user, look inside your database (either via pgAdmin or prisma studio) and take a screenshot of the new row generated when you logged in.

### PR DESCRIPTION
**Pull request by jessicasu77**
### ℹ️ Issue

Closes #11 

### 📝 Description

Uses Prisma Adapter to create a User in the database upon creating an account

- Prisma adapter creates a User, Session, Account, and VerificationToken (changes are in Schema.Prisma)

Note:
- A session is first created when a user logs in for the first time and recreated every time a session is expired
- A User can have many accounts and many sessions
- A User has a PrivilegeLevel which is Profile by default (we removed the Profile table from #6 that previously identified each user and their PrivilegeLevel since NextAuth creates a User for us with all information that was in the Profile table)

### ✔️ Verification

What steps did you take to verify your changes work? These should be clear enough for someone to be able to clone the branch and follow the steps themselves. 

Provide screenshots of any new components, styling changes, or pages. 

User table which has a PrivilegeLevel of Profile by default
C4CAuthFinal

Account table
C4CAccount

Session table
C4CSession

### 🏕️ (Optional) Future Work / Notes

Did you notice anything ugly during the course of this ticket? Any bugs, design challenges, or unexpected behavior? Write it down so we can clean it up in a future ticket!

@yanDavid21 and I think that it may be helpful to have an endpoint to make users admin (so that we can shorten the amount of time it takes to make people admin manually). This can be done by a PUT endpoint which changes the PrivilegeLevel of a User from its default value of Profile to Admin.

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -11,24 +11,66 @@ datasource db {
...
-  email          String         @unique
-  privilegeLevel PrivilegeLevel @default(PROFILE)
+  id        Int      @id @default(autoincrement())
+  title     String
+  content   String
+  createdAt DateTime @default(now())
 }
 
 enum PrivilegeLevel {
   PROFILE
```
Comment (rymaju) [inline prisma/schema.prisma:26]: ❓ I don't get this one so maybe someone can explain, why is privilege level one of "profile" or "admin"? I get "admin", but "profile"?? Can we change "profile" to "none"?

File: prisma/schema.prisma [hunk 2]
```diff
@@ -11,24 +11,66 @@ datasource db {
...
+
+model User {
+  id             String         @id @default(cuid())
+  name           String?
+  email          String?        @unique
+  emailVerified  DateTime?
+  image          String?
+  accounts       Account[]
+  sessions       Session[]
+  privilegeLevel PrivilegeLevel @default(PROFILE)
```
Comment (rymaju) [inline prisma/schema.prisma:75]: ❤️

**Review by rymaju — APPROVED** [review]
LGTM

File: prisma/schema.prisma (hunk 1 above)
Comment (varunthakkar1) [reply prisma/schema.prisma:26]: Yeah can we change this to 'basic' maybe? 'none' also works

**Review by varunthakkar1 — DISMISSED** [review]
Looks good! Just change the 'profile' enum name

File: prisma/schema.prisma (hunk 1 above)
Comment (jessicasu77) [reply prisma/schema.prisma]: yup just did!

File: pages/api/auth/[...nextauth].js [hunk 3]
```diff
@@ -1,19 +1,25 @@
...
       clientSecret: process.env.COGNITO_CLIENT_SECRET,
       issuer: process.env.COGNITO_ISSUER,
     }),
     // ...add more providers here
   ],
+  callbacks: {
+    async session({ session, _token, user }) {
+      return { ...session, user: { ...session.user, privilegeLevel: user.privilegeLevel } };
+    },
+  },
```
Comment (varunthakkar1) [inline pages/api/auth/[...nextauth].js:24]: 💯

File: README.md [hunk 4]
```diff
@@ -137,3 +137,24 @@ You can check out [the Next.js GitHub repository](https://github.com/vercel/next
 The easiest way to deploy your Next.js app is to use the [Vercel Platform](https://vercel.com/new?utm_medium=default-template&filter=next.js&utm_source=create-next-app&utm_campaign=create-next-app-readme) from the creators of Next.js.
 
 Check out our [Next.js deployment documentation](https://nextjs.org/docs/deployment) for more details.
+
+## Sessions
+
+### `useSession()`
```
Comment (varunthakkar1) [inline README.md:143]: 💯

**Review by varunthakkar1 — APPROVED** [review]
💯💯💯

**Review by rymaju — APPROVED** [review]
LGTM

**Pull request merged**